bf2fc6cc711aee1a0c2a/kas-fleet-manager
# feat( cluster_mgr): MGDSTRM-1384 Add logic to remove an OSD cluster when demand decreases

## Description

Closes https://issues.redhat.com/browse/MGDSTRM-1384

## Verification Steps


**To verify with using only one cluster:**
1. To verify, one can create a new cluster and add it to the local DB. 
2. Add the same cluster to the database again this time with an updated ( different ) ID field. The external cluster IDs may remain the same.
3. The reconciliation loop will detect the first addition to the database in a 'ready' state and call OCM to delete the cluster.
4. It will then soft delete the second from the database.
5. Finally it will remove dependencies 

**You can alternatively verify that the deprovisioning test is completing with success:**
```
OCM_ENV=development make test/integration TESTFLAGS="-run TestClusterManager_SuccessfulReconcileDeprovisionCluster"
```

## Type of change


## Checklist (Definition of Done)
